--- a/setup.py
+++ b/setup.py
@@ -280,6 +280,9 @@
     TORCH_MINOR = int(torch.__version__.split(".")[1])
     version = f"{version}+cu.{torch.version.cuda}.torch.{TORCH_MAJOR}.{TORCH_MINOR}"
 
+TORCH_MAJOR = int(torch.__version__.split(".")[0])
+TORCH_MINOR = int(torch.__version__.split(".")[1])
+
 setup(
     name='sageattention',
     version=version,
@@ -293,4 +296,7 @@
     python_requires='>=3.9',
     ext_modules=ext_modules,
     cmdclass=cmdclass,
+    install_requires=[
+        f"torch=={TORCH_MAJOR}.{TORCH_MINOR}.*",
+    ],
 )
